### PR TITLE
UCS/memtrack: resuscitation procedures

### LINKED ISSUE #394: memtrack doesn't work properly with shared memory
Using memtrack seems to fail the gtest tests.

for ud and rc:

```
$UCS_MEMTRACK_DEST=stdout ./test/gtest/gtest --gtest_filter=rc/test_uct_perf.envelope*

[vegas17:8969 :0]    memtrack.c:98   Fatal: memory allocation name too long: 'uct_rc_verbs_iface_t' (len: 20, max: 19)
[vegas17:8969 :1]    memtrack.c:98   Fatal: memory allocation name too long: 'uct_rc_verbs_iface_t' (len: 20, max: 19)

[vegas17:9007 :0]    memtrack.c:98   Fatal: memory allocation name too long: 'uct_ud_verbs_iface_t' (len: 20, max: 19)
[vegas17:9007 :1]    memtrack.c:98   Fatal: memory allocation name too long: 'uct_ud_verbs_iface_t' (len: 20, max: 19)
```

after changing the UCS_MEMTRACK_NAME_MAX  to 30 instead of 20, it fails on other issues:

```
$UCS_MEMTRACK_DEST=stdout ./test/gtest/gtest --gtest_filter=rc/test_uct_perf.envelope*
[     INFO ] Using random seed of 16758
Note: Google Test filter = rc/test_uct_perf.envelope*
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from rc/test_uct_perf
[ RUN      ] rc/test_uct_perf.envelope/0
[     INFO ] Not enforcing performance on SandyBridge far socket
[vegas17:9361 :0]    memtrack.c:150  Assertion `buffer->magic == UCS_MEMTRACK_MAGIC' failed
==== backtrace ====
 2 0x000000000003878a ucs_handle_error()  /labhome/alinas/workspace/ucx/src/ucs/debug/debug.c:630
 3 0x000000000003a313 __ucs_abort()  /labhome/alinas/workspace/ucx/src/ucs/debug/log.c:223
 4 0x000000000003aeee ucs_memtrack_record_release()  /labhome/alinas/workspace/ucx/src/ucs/debug/memtrack.c:150
 5 0x000000000003bd26 ucs_memtrack_releasing()  /labhome/alinas/workspace/ucx/src/ucs/debug/memtrack.c:517
 6 0x0000000000011cdf uct_ib_mem_free()  /labhome/alinas/workspace/ucx/src/uct/ib/base/ib_device.c:153
 7 0x000000000000f5ef uct_pd_mem_free()  /labhome/alinas/workspace/ucx/src/uct/tl/context.c:383
 8 0x000000000000fffc uct_mem_free()  /labhome/alinas/workspace/ucx/src/uct/tl/memory.c:156
 9 0x00000000000102f9 uct_iface_mem_free()  /labhome/alinas/workspace/ucx/src/uct/tl/memory.c:231
10 0x00000000004ce093 uct_perf_test_free_mem()  /labhome/alinas/workspace/ucx/src/tools/perf/libperf.c:112
11 0x00000000004d0713 uct_perf_cleanup()  /labhome/alinas/workspace/ucx/src/tools/perf/libperf.c:888
12 0x00000000004d0c28 ucx_perf_run()  /labhome/alinas/workspace/ucx/src/tools/perf/libperf.c:1010
13 0x000000000040dc79 _ZN9test_perf11thread_funcEPv()  /labhome/alinas/workspace/ucx/test/gtest/base/test_perf.cc:186
14 0x0000003848c079d1 start_thread()  pthread_create.c:0
15 0x00000038488e886d __clone()  ??:0
===================
Aborted
```

mm fails as well with:

```
[vegas17:9328 :0]    mm_iface.c:225  Assertion `iface->read_index <= iface->recv_fifo_ctl->head' failed
```

Also, need to review and fix the ucs_mmap implementation.

### PR DESCRIPTION
**Pull request by shssf**
Fixes #394

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1274/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1275/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3221/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1276/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3223/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3222/ for details (Mellanox internal link).

File: src/ucs/debug/memtrack.c [hunk 1]
```diff
@@ -325,6 +325,19 @@ int ucs_munmap(void *addr, size_t length)
...
 }
 
+char *ucs_strdup(const char *src)
+{
+  char *str;
+  size_t len = strlen(src);
+
+  str = ucs_malloc(len + 1, "ucs_strdup");
+  if (str) {
+    memcpy(str, src, len + 1);
```
Comment (yosefe) [inline src/ucs/debug/memtrack.c]: indentation should be 4 spaces

File: src/ucs/debug/memtrack.c [hunk 2]
```diff
@@ -325,6 +325,19 @@ int ucs_munmap(void *addr, size_t length)
                   length + sizeof(*buffer) + buffer->offset);
 }
 
+char *ucs_strdup(const char *src)
+{
+  char *str;
+  size_t len = strlen(src);
+
+  str = ucs_malloc(len + 1, "ucs_strdup");
```
Comment (yosefe) [inline src/ucs/debug/memtrack.c]: need to add "name" argument to ucs_strdup, and pass it to ucs_malloc, rather than passing "ucs_strdup"

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1277/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3225/ for details (Mellanox internal link).

**Pull request merged**